velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -209,4 +209,34 @@ struct LastDayFunction {
   }
 };
 
+template <typename T>
+struct DateSubFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Date>& result,
+      const arg_type<Date>& date,
+      const int32_t value) {
+    int32_t subValue = 0 - value;
+    result = addToDate(date, DateTimeUnit::kDay, subValue);
+    return true;
```

Comment:
Make this function return 'void' and remove this statement.

Reply:
Done